src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -170,21 +170,23 @@ public bool HasResponseStarted
             get { return _requestProcessingStatus == RequestProcessingStatus.ResponseStarted; }
         }
 
-        protected FrameRequestHeaders FrameRequestHeaders => _frameHeaders.RequestHeaders;
+        protected FrameRequestHeaders FrameRequestHeaders { get; private set; }
+
+        protected FrameResponseHeaders FrameResponseHeaders { get; private set; }
 
         public void InitializeHeaders()
         {
-            if (_frameHeaders == null)
+            if (FrameRequestHeaders == null)
             {
-                _frameHeaders = new Headers(ServerOptions);
-                RequestHeaders = _frameHeaders.RequestHeaders;
-                ResponseHeaders = _frameHeaders.ResponseHeaders;
+                RequestHeaders = FrameRequestHeaders = new FrameRequestHeaders();
```

Comment:
This looks kind of gross. Can we just change the type of the `RequestHeaders` and `ResponseHeaders` properties from `IHeaderDictionary` to `FrameRequestHeaders` and `FrameResponseHeaders` and use those properties everywhere?